.atomist/editors/HavePathExpressionEngine.ts

```diff
@@ -0,0 +1,40 @@
+import { EditProject } from '@atomist/rug/operations/ProjectEditor'
+import { Project } from '@atomist/rug/model/Project'
+import { PathExpression, PathExpressionEngine } from '@atomist/rug/tree/PathExpression'
+import { Pattern } from '@atomist/rug/operations/RugOperation'
+import { File } from '@atomist/rug/model/File'
+import { Editor, Parameter, Tags } from '@atomist/rug/operations/Decorators'
+
+@Editor("HavePathExpressionEngine", "get access to a path expression engine in a Rug editor")
+@Tags("documentation")
+class HavePathExpressionEngine implements EditProject {
+
+    @Parameter({
+        displayName: "Rug to modify",
+        description: "name of a TypeScript rug",
+        pattern: Pattern.any,
```

Comment:
Do we have a restriction on the names of rugs, and does that restriction have a name?